src/module.c

```diff
@@ -720,12 +724,16 @@ static int aliasAddCommon(RedisModuleCtx *ctx, RedisModuleString **argv, int arg
     return REDISMODULE_ERR;
   }
 
-  const char *alias = RedisModule_StringPtrLen(argv[1], NULL);
+  size_t length = 0;
+  const char *rawAlias = RedisModule_StringPtrLen(argv[1], &length);
+  HiddenName *alias = NewHiddenName(rawAlias, length, false);
   StrongRef alias_ref = IndexAlias_Get(alias);
+  int rc = REDISMODULE_OK;
   if (!skipIfExists || !StrongRef_Equals(alias_ref, ref)) {
-    return IndexAlias_Add(alias, ref, 0, error);
+    rc = IndexAlias_Add(alias, ref, 0, error);
   }
-  return REDISMODULE_OK;
+  HiddenName_Free(alias, false);
```

Comment:
Same - set in the alias-dictionary and then free ?

Reply:
Not sure what you meant regarding same.
Note the add function can fail.
It will only duplicate the alias if it doesn't already exist in the container.
So the caller should just always free.
In the rlookup branch it becomes references and the cost decreases.